### DISCUSSION TITLE
CloudFrontKeyValueStore Signature v4a

### DISCUSSION THREAD

**Discussion by djoleeeee** [aws/aws-sdk-php#2900] [Q&A] [answered]
```
        $client = new Aws\CloudFrontKeyValueStore\CloudFrontKeyValueStoreClient(array(
            'region'  => 'us-east-1',
            'version' => 'latest',
            'credentials' => ['key'=> 'XXX', 'secret'=>'XXX'],
            'signature_version' => 'v4a'
        ));
        
        $list = $client->listKeys([
            'KvsARN' => 'arn:aws:cloudfront::XXX:key-value-store/XXX', // REQUIRED
        ]);
```
        
 Error: This operation requests `sigv4a` auth schemes, but the client currently supports `sigv4`, `none`, `bearer`, `sigv4-s3express`.
 
 Can someone help me with this?

**Comment by yenfryherrerafeliz** [comment, marked as answer]
Hi @djoleeeee, the service client `CloudFrontKeyValueStoreClient` requires the awscrt extension to be installed in order to work. Please follow the instruction found [here](https://github.com/awslabs/aws-crt-php) to get it installed. Once you do this you should have not issue related to this. Please let me know if that helps.

Thanks!

**Reply by djoleeeee** [reply]
Thank you so much! 

I missed the PHP extension part (I just loaded the library with Composer). 

Now it works.

**Comment by djoleeeee** [comment]
This error occurs again after I upgraded my PHP version from 7.4 to 8.2.

Any ideas?

**Reply by yenfryherrerafeliz** [reply]
Hi @djoleeeee, I think it is because you would need to again install the awscrt extension but for that PHP version, which is 8.2. Could you please also confirm which SDK version you are using?

Thanks!

**Reply by djoleeeee** [reply]
I did it, of course.

I tried both methods: "pecl install awscrt" and manually building from source. 

Both times, everything went fine, and php -m lists awscrt in the extension list, but the code is still not working.

I tried the latest aws-crt version and 1.2.4.

I also tried with multiple aws-sdk versions, but without any success.

**Reply by yenfryherrerafeliz** [reply]
@djoleeeee Ok. Could you please confirm what is the version for the package `aws/aws-sdk-php` that you are using?. I just want to test it and see what could be happening, but would be helpful to know the specific version you are at.

Thanks!

**Reply by djoleeeee** [reply]
3.315.6

**Reply by djoleeeee** [reply]
@yenfryherrerafeliz any news?

**Reply by yenfryherrerafeliz** [reply]
@djoleeeee I will get back to you soon.

Thanks!

**Reply by stobrien89** [reply]
Hi @djoleeeee,

I was unable to reproduce using PHP version 8.2.  Is it possible your app is calling another instance of 8.2 somewhere else on your machine?  It would be helpful if you could provide debug logs (add `'debug' => true` to your client config).  Just make sure to obscure any sensitive information, such as account numbers.

**Reply by djoleeeee** [reply]
Suddenly, it worked after I downloaded the latest version of aws-sdk-php.

However, I don't think that's the reason. I suspect that I forgot to restart the server after the installation.

I'm sorry to bother you.

Thank you.

**Reply by stobrien89** [reply]
No bother at all— glad you were able to get it working!
